src/shared/src/components/molecules/AltinnModal.tsx

```diff
@@ -98,66 +92,53 @@ const styles = createStyles({
   },
 });
 
-export class AltinnModal extends React.Component<IAltinnModalComponentProvidedProps, IAltinnModalComponentState> {
-  public render() {
-    const { classes, printView } = this.props;
-    if (!printView) {
-      return (
-        <Modal
-          open={this.props.isOpen}
-          className={this.props.classes.scroll}
-          hideBackdrop={this.props.hideBackdrop}
-          onBackdropClick={this.props.allowCloseOnBackdropClick === false ? null : this.props.onClose}
-        >
-          <div className={classes.modal}>
-            <div className={classes.header}>
-              {this.props.hideCloseIcon && this.props.hideCloseIcon === true ? null :
-                <IconButton
-                  className={classNames(
-                    classes.iconBtn,
-                    { [classes.closeButtonOutsideModal]: this.props.closeButtonOutsideModal === true},
-                  )}
-                  onClick={this.props.onClose}
-                >
-                  <i tabIndex={0} className={classNames('ai ai-exit-test', classes.iconStyling)} />
-                </IconButton>
-              }
-              <Typography className={classes.headerText}>
-                {this.props.headerText}
-              </Typography>
-            </div>
-            <div className={classes.body}>
-              {this.props.children}
-            </div>
-          </div>
-        </Modal>
-      );
-    } else {
-      return (
-        <div className={classes.modal}>
-          <div className={classes.header}>
-            {this.props.hideCloseIcon && this.props.hideCloseIcon === true ? null :
-              <IconButton
-                className={classNames(
-                  classes.iconBtn,
-                  { [classes.closeButtonOutsideModal]: this.props.closeButtonOutsideModal === true},
-                )}
-                onClick={this.props.onClose}
-              >
-                <i tabIndex={0} className={classNames('ai ai-exit-test', classes.iconStyling)} />
-              </IconButton>
-            }
-            <Typography className={classes.headerText}>
-              {this.props.headerText}
-            </Typography>
-          </div>
-          <div className={classes.body}>
-            {this.props.children}
-          </div>
-        </div>
-      );
-    }
+const useStyles = makeStyles(styles);
+
+const StyledModal = (props: React.PropsWithChildren<IAltinnModalComponentProvidedProps>) => {
+  const classes = useStyles();
+  return (
+    <div className={classes.modal}>
+      <div className={classes.header}>
+        {props.hideCloseIcon && props.hideCloseIcon === true ? null :
+          <IconButton
+            className={classNames(
+              classes.iconBtn,
+              {[classes.closeButtonOutsideModal]: props.closeButtonOutsideModal === true},
+            )}
+            onClick={props.onClose}
+          >
+            <i className={classNames('ai ai-exit-test', classes.iconStyling)} tabIndex={-1}/>
```

Comment:
~~This doesnt seem entirely correct to me - this will remove the ability to navigate to this by tabbing. I would think moving the tabIndex property to the parent `IconButton` would be better?~~ 

Or maybe its not needed at all, since I guess `IconButton` uses a `button` under the hood anyway, and so it will by default be possible to navigate to it.

So `tabindex` could probably removed entirely in this case.

WDYT?

Reply:
I think you are right. My first intent was to remove ( see first commit ), but because I don't currently know how this affects the component in practice I wanted a second opinion. From what I could get by asking @lorang92 yesterday, the other case where `tabindex={0}` was used it was intentionally because of the need to use `focus()` on an non interactive element.
In this case I'm not sure, but left it if that was the case here as well.
WDYT?